src/matter/cluster/BasicCluster.ts

```diff
@@ -4,17 +4,43 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import { StringT, UnsignedIntT } from "../../codec/TlvObjectCodec";
+import { StringT, UnsignedIntT, BooleanT, Field, ObjectT } from "../../codec/TlvObjectCodec";
 import { AttributeSpec, ClusterSpec } from "./ClusterSpec";
 
+const CapabilityMinimaStructT = ObjectT({
+  caseSessionsPerFabric: Field(0, UnsignedIntT),
+  subscriptionsPerFabric: Field(1, UnsignedIntT),
+});
+
+/**
+ * This cluster provides attributes and events for determining basic information about Nodes, which supports both
+ * Commissioning and operational determination of Node characteristics, such as Vendor ID, Product ID and serial number,
+ * which apply to the whole Node. Also allows setting user device information such as location.
+ */
 export const BasicClusterSpec = ClusterSpec(
     0x28,
     "Basic",
     {
+        dataModelRevision: AttributeSpec(0, UnsignedIntT),
         vendorName: AttributeSpec(1, StringT),
         vendorId: AttributeSpec(2, UnsignedIntT),
         productName: AttributeSpec(3, StringT),
         productId: AttributeSpec(4, UnsignedIntT),
+        nodeLabel: AttributeSpec(5, StringT, ''),
+        location: AttributeSpec(6, StringT, 'XX'),
+        hardwareVersion: AttributeSpec(7, UnsignedIntT, 0),
+        hardwareVersionString: AttributeSpec(8, StringT),
+        softwareVersion: AttributeSpec(9, UnsignedIntT, 0),
+        softwareVersionString: AttributeSpec(10, StringT),
+        manufacturingDate: AttributeSpec(11, StringT ), // optional
```

Comment:
Thank you for your thoughts. I agree that it makes sense to change the field "properties" to a object because it is more future proof, but leaving the id and type separate parameters. (so your first approach)

I also think we need an "Optional type" or we need to completely leave the optional fields aside because as soon as I add it here it has effects e.g. in "Device.ts" that the attribute needs to be defined because the attributes type struct has that value and a definition ... so it is not really "optional" anymore.
I would see two options for now: 
* Yes I add optional as "comment" and after the PR is merged you add the typing support and adjust all that in one run. (but that might need Device.ts gets some "unneeded" attributed just because needed
* You add optional typing into master and I rebase and then directly use it in the PR
* You add optional typing into my PR and we so have it together.

What do you think?